tests/test_features.py

```diff
@@ -10,14 +15,6 @@
 except KeyError:
     pass
 
-import matplotlib
```

Comment:
> For instance, if you have a 3d array, and you're normalizing along axis=1, then 
> S[:, i, :] should be normalized all as one "vector". If this sounds strange, imagine that you first permuted the 0 and 1 axes, so that you have S_permute[i, :, :], and then flatten the last two axes so that you just have S_permute_flat[i, :]. Any normalization you do to S should be equivalent to the corresponding vector-wise normalization on the flattened version.

Okay, let's work this out. @nils-werner and me bent our heads around this for a couple of hours with the result that librosa is doing something different than you explain - or we are plain dumb :)

Example:

```python
S = np.zeros((3, 4, 5))
S[:, :, :] = 3.0
S[0, 0, 0] = 12.0

B = librosa.util.normalize(S, norm=np.inf, axis=2)
print(B)
```
yields

```
[[[ 1.    0.25  0.25  0.25  0.25]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]]

 [[ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]]

 [[ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]
  [ 1.    1.    1.    1.    1.  ]]] 
```

What we would expect based on your explanation is:

```
[[[ 1.    1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]]

 [[ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]]

 [[ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]
  [ 0.25  1.    1.    1.    1.  ]]]
```

What do you think?